src/Validator/Csrf.php

```diff
@@ -0,0 +1,308 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Session\Validator;
+
+use Laminas\Session\Container;
+use Laminas\Validator\AbstractValidator;
+
+use function assert;
+use function explode;
+use function is_array;
+use function is_string;
+use function md5;
+use function random_bytes;
+use function sprintf;
+use function str_replace;
+use function strtr;
+
+/**
+ * @psalm-type OptionsArgument = array{
+ *     name?: non-empty-string,
+ *     salt?: non-empty-string,
+ *     session?: Container,
+ *     timeout?: int,
+ * }
+ */
+final class Csrf extends AbstractValidator
+{
+    /**
+     * Error codes
+     *
+     * @const string
+     */
+    public const NOT_SAME = 'notSame';
+
+    /**
+     * Error messages
+     *
+     * @var array<string, string>
+     */
+    protected array $messageTemplates = [
```

Comment:
Do we have rules for visibility in final classes?

Reply:
I've made everything private except for this variable which is accessed by `AbstractValidator`